ports/rp2/memmap_mp.ld

```diff
@@ -248,11 +248,19 @@ SECTIONS
     __StackTop = ORIGIN(SCRATCH_Y) + LENGTH(SCRATCH_Y);
     __StackOneBottom = __StackOneTop - SIZEOF(.stack1_dummy);
     __StackBottom = __StackTop - SIZEOF(.stack_dummy);
+    /* Define start and end of GC heap */
+    __GcHeapStart = __bss_end__;
+    __GcHeapEnd   = __StackLimit;
     PROVIDE(__stack = __StackTop);
 
     /* Check if data + heap + stack exceeds RAM limit */
     ASSERT(__StackLimit >= __HeapLimit, "region RAM overflowed")
 
+    /* Check GC heap is at least 40 KB */
+    /* This an arbitrary number. Really small programs could run in 20 KB. */
+    /* Using all RAM available it will about 230 KB for PICO and 196 KB for PICO_W */
+    ASSERT((__GcHeapEnd - __GcHeapStart) > 40*1024, "GcHeap is too small")
```

Comment:
I think we can make this 128k.  If there's less than that available then something went wrong somewhere, so it would be good to warn in such a case.

Reply:
I undertook this work because I want to turn off rp2040 SRAM that I don't need for my app.
This will reduce power consumption.
Others might want to as well. 
I am currently running using only 128 KBytes (out of a total of 264 KBytes).

But, in an "official" build there should be more than 128Kbytes of GC space.
So OK.

Reply:
I see... is 128kbytes the minimum you can tune it down to?  If so then we can accommodate that and make this check a bit smaller.  Does  64k work?

Reply:
Yes, the PICO_W has 6 banks of SRAM. 
Four are 64 KBytes and two are 4KBytes.

Normally the 4 KByte blocks are used for processor 0 and 1 stacks.
I am currently running a small micropython program in 128 KBytes with the last 8 KBytes used for stacks.
In this case the GC heap is about 56 KBytes.

I already pushed a commit for 128 KBytes. 

Someone wanting to build micropython for less RAM will have to change the MEMORY declaration at line 24 in 
memmap_mp.ld.
They will also need to change the assert. Not a big deal.

I suppose I could add a comment above the "MEMORY" declaration explaining that.

Reply:
> Someone wanting to build micropython for less RAM will have to change the MEMORY declaration at line 24 in
memmap_mp.ld.
They will also need to change the assert. Not a big deal.

That's true, they'll need to modify things anyway.  And the initial warning will let them know the heap has shrunk, which would be a good indication that their change to the MEMORY declaration was actually working.

So let's leave it at 128k then.